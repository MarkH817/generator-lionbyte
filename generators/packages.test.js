const { getPackages } = require('./packages')

describe('getPackages', () => {
  test('base node', () => {
    const result = getPackages({ projectType: 'node' })
    expect(result).toMatchSnapshot()
  })

  test('base node + git hooks', () => {
    const result = getPackages({ gitHooks: true, projectType: 'node' })
    expect(result).toMatchSnapshot()
  })

  test('base frontend', () => {
    const result = getPackages({
      projectType: 'frontend',
      react: false
    })
    expect(result).toMatchSnapshot()
  })

  test('base frontend + react', () => {
    const result = getPackages({
      projectType: 'frontend',
      react: true
    })
    expect(result).toMatchSnapshot()
  })
})
