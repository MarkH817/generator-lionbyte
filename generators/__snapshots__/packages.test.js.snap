// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPackages base frontend + react 1`] = `
Object {
  "dependencies": Array [
    "core-js@3",
    "react",
    "react-dom",
    "regenerator-runtime",
  ],
  "devDependencies": Array [
    "@babel/core",
    "@babel/plugin-syntax-dynamic-import",
    "@babel/preset-env",
    "@babel/preset-react",
    "@lionbyte/eslint-config",
    "@types/jest",
    "@types/node",
    "@types/react",
    "@types/react-dom",
    "autoprefixer",
    "babel-jest",
    "babel-loader",
    "babel-plugin-dynamic-import-node",
    "clean-webpack-plugin",
    "css-loader",
    "cssnano",
    "eslint",
    "html-webpack-plugin",
    "jest",
    "less",
    "less-loader",
    "mini-css-extract-plugin",
    "postcss-loader",
    "prettier",
    "style-loader",
    "typescript",
    "webpack",
    "webpack-cli",
    "webpack-dev-server",
    "webpack-merge",
  ],
}
`;

exports[`getPackages base frontend 1`] = `
Object {
  "dependencies": Array [
    "core-js@3",
    "regenerator-runtime",
  ],
  "devDependencies": Array [
    "@babel/core",
    "@babel/plugin-syntax-dynamic-import",
    "@babel/preset-env",
    "@lionbyte/eslint-config",
    "@types/jest",
    "@types/node",
    "autoprefixer",
    "babel-jest",
    "babel-loader",
    "babel-plugin-dynamic-import-node",
    "clean-webpack-plugin",
    "css-loader",
    "cssnano",
    "eslint",
    "html-webpack-plugin",
    "jest",
    "less",
    "less-loader",
    "mini-css-extract-plugin",
    "postcss-loader",
    "prettier",
    "style-loader",
    "typescript",
    "webpack",
    "webpack-cli",
    "webpack-dev-server",
    "webpack-merge",
  ],
}
`;

exports[`getPackages base node + git hooks 1`] = `
Object {
  "dependencies": Array [],
  "devDependencies": Array [
    "@lionbyte/eslint-config",
    "@types/jest",
    "@types/node",
    "eslint",
    "husky",
    "jest",
    "lint-staged",
    "prettier",
    "typescript",
  ],
}
`;

exports[`getPackages base node 1`] = `
Object {
  "dependencies": Array [],
  "devDependencies": Array [
    "@lionbyte/eslint-config",
    "@types/jest",
    "@types/node",
    "eslint",
    "jest",
    "prettier",
    "typescript",
  ],
}
`;
